### ISSUE TITLE
Texture replacements load incorrectly after sufficient playtime

### ISSUE THREAD

**Issue by Zynetic** [hrydgard/ppsspp#16687]
### Game or games this happens in

NPJH50332 - Phantasy Star Portable 2 Infinity

### What area of the game

This issue occurs anywhere where texture replacements are being used. With it off, or where textures are not being replaced, there are no issues. Note that it doesn't happen immediately, only after some time has passed - there doesn't seem to be a set amount of time / writes / etc. for it to happen.

### What happens

After a sufficient amount of playtime, texture replacements will start loading into incorrect places. I thought this was related to #16642, but it appears not to have been resolved in V1.14.2.

An example can be found below:

![NPJH50332_00000](https://user-images.githubusercontent.com/33199242/210140395-bae394a8-bbea-41e5-88c3-fbaaf501ceb7.png)

### What should happen

The below image isn't with texture replacement, but it is very similar to how it should look:

![2022-12-31 14-37-28](https://user-images.githubusercontent.com/33199242/210140405-1c03fbe7-d073-4372-afbb-6a38ab33a945.png)

It's probably worth mentioning this is with the unofficial english patch installed, but I can't imagine it could be the culprit.

### Platform

Linux / BSD

### Mobile phone model or graphics card

Steam Deck / SteamOS

### PPSSPP version affected

V1.14.2

### Last working version

V1.13.2

### Graphics backend (3D API)

OpenGL / GLES

### Checklist

- [X] Test in the [latest git build](https://buildbot.orphis.net/ppsspp/) in case it's already fixed.
- [X] [Search for other reports](https://github.com/hrydgard/ppsspp/search?q=my+issue&type=issues) of the same issue.
- [X] Try resetting settings or older versions and include if the issue is related.
- [X] Try changing graphics settings to determine if one causes the glitch (especially speed hacks or enhancements/replacements.)
- [X] Include logs or screenshots of issue.

**Comment by hrydgard** [comment]
* Are you using a zipped texture replacement pack?
* Have you tried the very latest builds: https://buildbot.orphis.net/ppsspp/ ?

**Comment by hrydgard** [comment]
Ok, if standalone then it's some different problem than the one recently fixed with zip files.

**Comment by unknownbrackets** [comment]
It's hard to tell if this is just a poorly implemented texture pack (which might be harsh given that a lot of short-sighted shortcuts have been added to texture replacement.)  It's actually very easy to reduce the hashed bytes and ignore cluts and etc. in a way that will drastically increase hash collisions, and because of how PSP games treat stride, people commonly use these shortcuts.

I think we'd want an instance of this happening without such shortcuts to be sure and to be able to replicate.

-[Unknown]

**Comment by eleriaqueen** [comment]
Hi, I'm the author of one of the Phantasy Star Portable 2 - Infinity texture packs, namely the HD UI one, but I haven't touched PPSSPP for almost a year.

Here are the settings my pack is using : 
`[options]
version = 1
hash = xxh64
ignoreAddress = true
ignoreMipmap = false`

I don't remember why I chose to "ignoreAddress", might have been because the same texture appeared in several places... idk

**Comment by unknownbrackets** [comment]
If you're not using "reduceHash" then it's probably not that.  Is this only with PNGs, or with ZIMs too?  Does it happen using a zip?  If you could find a git build it started happening in, that'd help.

-[Unknown]

**Comment by eleriaqueen** [comment]
My releases are only PNGs so far.

**Comment by unknownbrackets** [comment]
Is this perhaps only happening on Linux then?  How long is "sufficient" playtime?

-[Unknown]

**Comment by unknownbrackets** [comment]
Does it only happen with OpenGL/GLES then?  Maybe #16756.

-[Unknown]

**Comment by eleriaqueen** [comment]
Ohhh, my two attempts at triggering the issue were both on Vulkan backend.

**Comment by hrydgard** [comment]
Hm, this is unrelated, but I'm surprised to hear that Vulkan would have some specific performance issues in this game. Is it slow throughout or at some specific times?

**Comment by hrydgard** [comment]
Hm, might be some AMD-specific thing, I don't currently have any AMD GPU devices (though probably should get one..). Weird anyway - I normally expect the Vulkan backend be the best or very close to it at everything.

**Comment by hrydgard** [comment]
Also can you try the latest builds, v1.14.4-128 or later? As Unknown says, it's possible that #16756 fixed it.

**Comment by eleriaqueen** [comment]
I know I'm a bit late, but yeah the issue maybe only happens on AMD hardware, 'cause I triggered it on PPSSPP 1.14.3 on my steamdeck after doing only two missions with my textures pack on and getting back to the "base". I'm unsure how to test 1.14.4-128 or later on my deck though as BuildBot doesn't produce Linux builds. 
Does 1.14.4 stable include the fix you mention ? I just updated to that version through the package manager.

**Comment by unknownbrackets** [comment]
Thanks for checking, I'll go ahead and close this - if you see it again in the same way, we can reopen.  But if you see it again in another later version much later, it could just be a new issue, so it'd be better to create a new issue.

-[Unknown]
